<launch>
  <machine name="lime" address="lime" default="false" user="muku" />

  <arg name="INPUT" default="/hsrb/head_rgbd_sensor" />
  <arg name="INPUT_ON" default="false" />
  <arg name="NET" default="zf" />

  <group if="$(arg INPUT_ON)">
    <include file="$(find openni2_launch)/launch/openni2.launch">
      <arg name="depth_registration" value="True" />
    </include>
  </group>

  <node name="faster_rcnn" machine="lime"
        pkg="jsk_perception" type="faster_rcnn_caffenet.py" > 
    <remap from="~input" to="$(arg INPUT)/rgb/image_rect_color" />
    <param name="net" value="$(arg NET)" />
  </node>

  <node name="draw_rect_array"
        pkg="jsk_perception" type="draw_rect_array.py">
    <remap from="~input" to="$(arg INPUT)/rgb/image_rect_color" />
    <remap from="~input/rect_array" to="faster_rcnn/rect_array" />
    <rosparam>
      approximate_sync: True
      slop: 10
    </rosparam>
  </node>

  <node name="image_view"
        pkg="image_view" type="image_view">
    <remap from="image" to="draw_rect_array/output" />
  </node>

  <node name="recognised_tf_publisher"
        pkg="jsk_perception" type="tf_publisher_from_recognition.py">
    <remap from="~input/camera_info" to="$(arg INPUT)/rgb/camera_info" />
    <remap from="~input/depth" to="$(arg INPUT)/depth_registered/image_raw" />
    <remap from="~input/rect_array" to="/faster_rcnn/rect_array" />
    <remap from="~input/recog" to="/faster_rcnn/output" />
  </node>

</launch>
